Auto respond with "%name% has chat disabled" if chat disabled.

rapito opened the pull request:
### Auto respond with "%name% has chat disabled" if chat disabled.
This only happens on singles, on Teams it wont come up. 
![image](https://user-images.githubusercontent.com/1380104/111228316-60d64b00-85ba-11eb-88aa-5b6f641a6311.png)

##### Bug Fixes: 
- Prevent messages on same frame to hide each other
- Japenese names now work as expected
- Players in Teams can now chat without issues

rapito committed:
If Quick Chat is disabled, auto respond with "I have chat disabled" message
+ Prevent messages on same frame to hide each other

rapito committed:
Autoformat

rapito committed:
Autoformat

JLaferri commented:
Didn't look very hard but I didn't see it in my skim: does this only happen in singles? Probably shouldn't get sent in teams lobbies.

rapito committed:
Prevent CHAT IS DISABLED message if on teams.

rapito commented:
> Didn't look very hard but I didn't see it in my skim: does this only happen in singles? Probably shouldn't get sent in teams lobbies.

_Fixed._

rapito committed:
Make sure to set proper player index when sending a chat message.
- Fixes bug where users are no longer able to send chat messages after a couple are sent while in TEAMS mode

rapito commented:
@JLaferri This PR now also includes the fix for users on Teams not being able to send chat messages after a couple of them are sent. 
(https://github.com/project-slippi/Ishiiruka/pull/256/commits/213de1adc6227637f8f528749260d6598a323d33)

I also removed the logs you mentioned @NikhilNarayana

rapito committed:
Encode player names and chat message to utf32 and compare matching characters with integers when building Premade Text Data
+ Refactor Map to force each char to UTF32 encoding.
Fixes this issue: https://twitter.com/Kounotori_ssbm/status/1370050876744769538

rapito commented:
This PR now includes a patch that fixes Japanese Names when using Quick Chat.

 https://twitter.com/Kounotori_ssbm/status/1370050876744769538
 
![image](https://user-images.githubusercontent.com/1380104/111093073-53fc1d80-850e-11eb-9e01-2655f0a84d8c.png)

rapito committed:
Change message with a new premade text that says "%name has chat disabled"
+ Restore a line that makes this whole thing work

rapito committed:
Merge remote-tracking branch 'origin/release/2.3.0' into release/2.3.0

rapito committed:
Set proper encoding to file

JLaferri committed:
Merge branch 'release/2.3.0' of https://github.com/project-slippi/Ishiiruka into pr/256

JLaferri commented on Source/Core/Core/Slippi/SlippiPremadeText.h:
```diff
@@ -1,5 +1,6 @@
-#pragma once
+﻿#pragma once
 
+#include "../../Common/StringUtil.h"
```
is a relative path needed? dont remember seeing this anywhere else this is included uses `#include "Common/StringUtil.h"`

JLaferri committed:
remove relative import path
